### PR TITLE
Add Cloud Speech sample.

### PR DESCRIPTION
**Pull request by jmdobry**

### CODE REVIEW

**Comment by JustinBeckwith** [comment]
I feel a little icky about including the binaries in the git repo.  Opinions on git-lfs for this use?

**Comment by jmdobry** [comment]
My opinion is to leave them in the git repo. That's how the other <lang>-docs-samples repos do it.

**Comment by jonparrott** [comment]
binary files are fine here.

@jerjou to review.

File: speech/recognize.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+'use strict';
+
+var google = require('googleapis');
+var async = require('async');
+var fs = require('fs');
+var url = 'https://speech.googleapis.com/$discovery/rest';
+
+function auth(callback) {
+  google.auth.getApplicationDefault(function (err, authClient) {
```
Comment (jonparrott) [inline speech/recognize.js]: I believe speech requires a service account for the time being.

**Comment by jonparrott** [comment]
LGTM, but please let jerjou take a look before merging.

File: speech/recognize.js (hunk 1 above)
Comment (jmdobry) [inline speech/recognize.js]: I got it to work both with an API key and letting it pick credentials.

**Comment by jmdobry** [comment]
This is basically the Node.js version of: https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/speech/api/speech_rest.py

File: speech/recognize.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
+// Copyright 2015-2016, Google, Inc.
```
Comment (jerjou) [inline speech/recognize.js]: I think this can just be 2016, no? That's when you wrote the sample?

File: test/speech/recognize.test.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+var test = require('ava');
```
Comment (jerjou) [inline test/speech/recognize.test.js]: 'ava'? Looks like 'java' without the 'j'. I'm assuming this works? :)

**Comment by jerjou** [comment]
`quit.raw` was for the streaming sample (so not necessary here for the batch one).
Also - looks like there's a vision png in this PR?

**Comment by jerjou** [comment]
Otherwise, LGTM

File: test/speech/recognize.test.js (hunk 3 above)
Comment (jmdobry) [inline test/speech/recognize.test.js]: Yes, this is correct.

**Comment by jmdobry** [comment]
Hopefully I can get a streaming example added soon.

**Pull request merged**